Tekton demo improvements

mpryc opened the pull request:
Couple of changes:
 1. Tekton demo can ran from local source tree (without -g flag).
 2. Running with -g flag ensures modifications to the pelorus git project happens in the temporary directory and not local source tree
 3. Better handling of branches:
    - if remote branch exists, use it
    - if remote branch does not exists, create one
 4. Cleanup function added.
 5. Ensure we don't pollute master branch.

Signed-off-by: Michal Pryc <mpryc@redhat.com>

## Testing Instructions

Local directory run:
clone pelorus fork, run demo on existing branch or non-exising, e.g.
```
git clone git@github.com:mpryc/pelorus.git
cd pelorus
./demo-tekton.sh -r non_existing_remote_branch -c 3
# Create new branch existing_remote_branch using GitHub UI (to ensure script works fine with remote branches not synced to current clone):
./demo-tekton.sh -r existing_remote_branch -c 3
```

Temp directory
- This is better as it works without polluting any of current files/branches from which script is being ran
- Ensure -g url is provided to location where local user have commit access, most likely https is not the one...
```
cd pelorus
./demo-tekton.sh -r non_existing_remote_branch -c 3 -g git@github.com:mpryc/pelorus.git
# Create new branch existing_remote_branch using GitHub UI (to ensure script works fine with remote branches not synced to current clone):
./demo-tekton.sh -r existing_remote_branch -c 3 -g  git@github.com:mpryc/pelorus.git
```

Failing master, we really don't want to run this script with master branch...
```
./demo-tekton.sh -r master -c 3 -g  git@github.com:mpryc/pelorus.git
```


@redhat-cop/mdt

weshayutin commented:
hrm.. still have a bug on the pelorus test.. unrelated to this PR
```
Test that Pelorus has found the latest git commit:
Build Type is buildconfig
test buildconfig: requires exporter named committime-exporter 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  2674  100  2674    0     0  15916      0 --:--:-- --:--:-- --:--:-- 15916
```

mpryc commented:
> There is quite a delay in the clean up. I think folks _may_ think the script is hung. Suggestion is to log the cleanup as it works through or add a message it may take 2 minutes or so. :)
> 
> ```
> /var/tmp/PELORUS/pelorus1/demo
> 1. Namespace 'demoapp1' already exists
> Clean up resources prior to execution:
> ```

@weshayutin added some printf statements to be more informative what's happening.

weshayutin commented:
```
./demo-tekton.sh -b buildconfig -n demoapp1 -a demoapp1 -g https://github.com/weshayutin/pelorus.git  -r demoapp1  -c 12 -t 5
```
The current_branch is not substituted in the buildConfig in 05-

weshayutin commented:
--- /tmp/meld-tmp3z5luvk2.yaml
+++ /home/whayutin/OPENSHIFT/git/KONVEYOR/PELORUS/upstream/pelorus/demo/tekton-demo-setup/05-build-and-deploy.yaml
@@ -22,8 +22,8 @@
       source:
         type: Git
         git:
-          uri: https://github.com/konveyor/pelorus
-          ref: master
+          uri: ${url}
+          ref: refs/heads/${APPLICATION_NAME}
         contextDir: "demo/python-example"
       strategy:
         sourceStrategy:

mpryc commented:
> We would have to make sure we sync the branch name w/ the app name for this to work.. perhaps a fix
> 
> ```
> --- /tmp/meld-tmp3z5luvk2.yaml
> +++ /home/whayutin/OPENSHIFT/git/KONVEYOR/PELORUS/upstream/pelorus/demo/tekton-demo-setup/05-build-and-deploy.yaml
> @@ -22,8 +22,8 @@
>        source:
>          type: Git
>          git:
> -          uri: https://github.com/konveyor/pelorus
> -          ref: master
> +          uri: ${url}
> +          ref: refs/heads/${APPLICATION_NAME}
>          contextDir: "demo/python-example"
>        strategy:
>          sourceStrategy:
> ```

Reworked a bit to include such. It's actually in 2 places, first is the template which takes the URI and REF and then it translates into tekton parameters.

Also added check if the Pelorus installation is suitable to run the demo (commit time properly monitoring namespace where we are building app).

mpryc commented:
/test 4.9-e2e-openshift

weshayutin commented:
![Screenshot from 2022-09-28 11-03-02](https://user-images.githubusercontent.com/138787/192844745-cde3c489-6495-4b49-84b7-420c3bf86678.png)

multiple builds working on binary :) woot thanks!
buildConfig and s2i working as designed as well..

Tested w/
# demo apps
./demo-tekton.sh -b buildconfig -n demoapp1 -a demoapp1 -g https://github.com/weshayutin/pelorus.git  -r demoapp1  -c 12 -t 5
./demo-tekton.sh -b binary -n demoapp2 -a demoapp2 -g https://github.com/weshayutin/pelorus.git  -r demoapp2  -c 12 -t 5
./demo-tekton.sh -b s2i -n demoapp3 -a demoapp3 -g https://github.com/weshayutin/pelorus.git  -r demoapp3  -c 12 -t 5
